subsys/bluetooth/host/classic/sdp.c

```diff
@@ -2230,84 +2277,134 @@ static void sdp_client_disconnected(struct bt_l2cap_chan *chan)
 	 * resets autonomous.
 	 */
 	(void)memset(&session->reqs, 0,
-		     sizeof(*session) - sizeof(session->chan));
+		     sizeof(*session) - offsetof(struct bt_sdp_client, reqs));
```

Comment:
The code comment above doesn't seem valid after the change. In two other places you're also doing something different: 
```c
(void)memset(&session->chan, 0,
		sizeof(*session) - offsetof(struct bt_sdp_client, chan));
```

This playing around with offsets, pointers and memset will quickly become a potential source of bugs in addition to also being hard to follow. I'd recommend rethinking this, perhaps add some well documented helper functions, or do the resetting of the object using some more explicit method than memset.

Reply:
Updated with explicit function, please help to review again.